src/commands/updateMark.ts

```diff
@@ -0,0 +1,35 @@
+import { CommandFactory } from "../core/types.js"
+import { EditorState, Transaction } from "prosemirror-state"
+import { MarkType } from "prosemirror-model"
+import { getMarkRange } from "./getMarkRange.js"
+
+export const updateMark: CommandFactory =
+  (
+    type: MarkType,
+    attrs: Object,
+    options: {
+      toExtent: boolean
+    } = { toExtent: false }
+  ) =>
+  (state: EditorState, dispatch: (tx: Transaction) => void) => {
+    const { tr, selection, doc } = state
+    let { from, to } = selection
+    const { $from, empty } = selection
+
+    if (empty || options.toExtent) {
```

Comment:
just flagging the empty prop from EditorState to be used in a hasContent() util we were thinking about